openebs/pkg/v1/maya_api.go

```diff
@@ -55,15 +55,40 @@ type OpenEBSVolume struct{}
 func (v OpenEBSVolume) GetMayaClusterIP(client kubernetes.Interface) (string, error) {
 	clusterIP := "127.0.0.1"
 
+	glog.Info("Checking maya-apiserver status")
+
 	//Fetch the Maya ClusterIP using the Maya API Server Service
 	sc, err := client.CoreV1().Services("default").Get("maya-apiserver-service", metav1.GetOptions{})
 	if err != nil {
-		glog.Fatalf("Error getting maya-api-server IP Address: %v", err)
+		glog.Errorf("Error getting maya-apiserver IP Address: %v", err)
 	}
 
 	clusterIP = sc.Spec.ClusterIP
-	glog.Infof("Maya Cluster IP: %v", clusterIP)
+	glog.Infof("maya-apiserver Cluster IP: %v", clusterIP)
+
+	var url bytes.Buffer
 
+	addr := clusterIP
+
+	url.WriteString("http://" + addr + ":5656" + "/latest/meta-data/instance-id")
+	resp, err := http.Get(url.String())
+
+	if err != nil {
+		glog.Errorf("Error while connecting maya-apiserver %s", err)
+		time.Sleep(5 * time.Second)
+		return v.GetMayaClusterIP(client)
+	}
+
+	defer resp.Body.Close()
+	body, err := ioutil.ReadAll(resp.Body)
+
+	if err != nil {
+		glog.Errorf("maya-apiserver is not available: Error %s , HTTP Status Code: %d", err, resp.StatusCode)
```

Comment:
Yes! I was able to test this. You can find the `openebs-provisioner` logs here: https://github.com/openebs/openebs/issues/245#issuecomment-339559751

Reply:
The earlier message says Error while connecting to maya-apiserver. This might actually indicate unavailability of maya api server i.e. 500 error code. However crossing that hurdle & getting an error in a ioutil operation does not indicate unavailability of maya server. Can the code check in terms of http status codes !!! May be in a different PR.

Reply:
Agree with this! provisioner should return an HTTP error code.  Issue for this https://github.com/openebs/openebs/issues/728